### PR TITLE
fixes #18853: provide support of array attributes in multipart/form-data request for axios and fetch typescript clients

### LINKED ISSUE #18853: [BUG][Typescript][Axios/Fetch] Wrong generated code for array of items using FormData (multipart/form-data)
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

I use open api spec 3 to define api to define two attributes : 

* mediasToAdd which is an array of files to upload

* mediasToDelete which is an array of files uuids to delete

-------

I use a Typescript Axios in one project and Typescript Fetch in another project,

Theses projects are consuming almost the same (PHP) API endpoints.

But the typescript-axios and typescript-fetch generators generates a code that does not build properly theses array attributes.

This is the generated code : 

```typescript
# For typescript axios
formParams.append('mediasToAdd', element as any);
formParams.append('mediasToDelete', element as any);
# For typescript fetch
localVarFormParams.append('mediasToAdd', element);
localVarFormParams.append('mediasToDelete', element);
```
This parameters is not considered as array when submitting to a PHP API.

But it should be : 

```typescript
# For typescript axios
formParams.append('mediasToAdd[]', element as any);
formParams.append('mediasToDelete[]', element as any);
# For typescript fetch
localVarFormParams.append('mediasToAdd[]', element);
localVarFormParams.append('mediasToDelete[]', element);
```
the consequence is that on API side theses values are received as simple attributes instead as arrays.

-------

Following the previous screenshot the API will receive : 

* mediasToAdd as a single binary (latest value provided)

* mediasToDelete as a single string (latest value provided)

##### openapi-generator version

I use exactly @openapitools/openapi-generator-cli@1.0.18-5.0.0-beta2

##### OpenAPI declaration file content or url

Capture d’écran 2024-04-24 à 17 54 45

##### Suggest a fix

The problem is located in theses files => 

* https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache
* https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/typescript-fetch/apis.mustache

### PR DESCRIPTION
**Pull request by jeandonaldroselin**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Please review my PR :
@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @topce @akehir @petejohansonxo @amakhrov @davidgamero @mkusaka

### CODE REVIEW

**Review by macjohnny — APPROVED** [review]

**Comment by macjohnny** [comment]
@jeandonaldroselin can you merge the most recent master and fix the CI issues if they still occur?

**Comment by jeandonaldroselin** [comment]
> @jeandonaldroselin can you merge the most recent master and fix the CI issues if they still occur?

Hello @macjohnny,
I will check the state ASAP and push the changes 👍🏽

**Comment by jeandonaldroselin** [comment]
> @jeandonaldroselin can you merge the most recent master and fix the CI issues if they still occur?

Hello @macjohnny,

Sorry I was busy a lot since last week.
I updated my PR with the latest master and updated the tests.
The related build just passed :).

Please can you have a look at it ?

File: samples/client/echo_api/typescript-axios/build/api.ts [hunk 1]
```diff
@@ -615,7 +615,7 @@ export const BodyApiAxiosParamCreator = function (configuration?: Configuration)
 
             if (files) {
                 files.forEach((element) => {
-                  localVarFormParams.append('files', element as any);
+                    localVarFormParams.append('files', element as any);
```
Comment (macjohnny) [inline samples/client/echo_api/typescript-axios/build/api.ts:618]: this looks unrelated

Comment (jeandonaldroselin) [reply samples/client/echo_api/typescript-axios/build/api.ts:618]: Hello,

Yes it looks unrelated.
I had the build failing and I had to run 
the command : 

bash bin/generate-samples.sh 

after this run, this command only added this space change.
